apps/scan/src/test/scala/org/lfdecentralizedtrust/splice/store/db/AcsSnapshotStoreTest.scala

```diff
@@ -543,6 +551,39 @@ class AcsSnapshotStoreTest
         }
       }
 
+      "fail if too many contracts were returned by HardLimit" in {
+        val owner = providerParty(1)
+        val holder = providerParty(2)
+        val amulet1 = lockedAmulet(owner, 10, 1L, 0.5)
+        val amulet2 = lockedAmulet(owner, 10, 2L, 0.5)
+        (for {
+          updateHistory <- mkUpdateHistory()
+          store = mkStore(updateHistory)
+          _ <- ingestCreate(
+            updateHistory,
+            amulet1,
+            timestamp1.minusSeconds(10L),
+            Seq(owner, holder),
+          )
+          _ <- ingestCreate(
+            updateHistory,
+            amulet2,
+            timestamp1.minusSeconds(10L),
+            Seq(owner, holder),
+          )
+          _ <- store.insertNewSnapshot(None, DefaultMigrationId, timestamp1)
+          _result <- store.getHoldingsState(
+            DefaultMigrationId,
+            timestamp1,
+            None,
+            HardLimit.tryCreate(1),
+            NonEmptyVector.of(owner),
+          )
+        } yield fail("should not get here, call should've failed")).failed.futureValue shouldBe a[
```

Comment:
Why do you need the `fail()` part, instead of something like `yield succeed`? Would `.failed.futureValue` not fail the test with a useful description if it didn't throw an exception?